addons/l10n_jo_edi/tests/jo_edi_common.py

```diff
@@ -0,0 +1,105 @@
+from odoo import Command
+from odoo.tools import misc
+from odoo.addons.account_reports.tests.common import TestAccountReportsCommon
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+
+
+class JoEdiCommon(AccountTestInvoicingCommon):
+    @classmethod
+    @TestAccountReportsCommon.setup_country('jo')
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.company_data['company'].write({
+            'name': 'Jordan Company',
+            'vat': '8000514',
+        })
+
+        def _create_tax(amount, amount_type):
+            return cls.env['account.tax'].create(
+                {
+                    'name': f'{amount_type} {amount}',
+                    'amount_type': amount_type,
+                    'amount': amount,
+                    'company_id': cls.company_data['company'].id,
+                    'include_base_amount': amount_type == 'fixed',
+                    'is_base_affected': amount_type == 'percent',
+                    'sequence': 2 if amount_type == 'percent' else 1,
+                })
+
+        cls.jo_general_tax_10 = _create_tax(10, 'percent')
+        cls.jo_special_tax_10 = _create_tax(10, 'fixed')
+        cls.jo_special_tax_5 = _create_tax(5, 'fixed')
+        cls.jo_general_tax_16_included = _create_tax(16, 'percent')
+        cls.jo_general_tax_16_included.price_include_override = 'tax_included'
+
+        cls.partner_jo = cls.env['res.partner'].create({
+            'name': 'Ahmad',
+            'ref': 'Jordan Partner',
+            'city': 'Amman',
+            'vat': '54321',
+            'zip': '94538',
+            'country_id': cls.env.ref('base.jo').id,
+            'state_id': cls.env.ref('base.state_jo_az').id,
+            'phone': '+962 795-5585-949',
+            'company_type': 'company',
+        })
+
+        # The rate of 1 USD = 2 JOD is meant to simplify tests
+        cls.usd = cls.env.ref('base.USD')
+        cls.setup_currency_rate(cls.usd, 0.5)
+
+    @classmethod
+    def setup_currency_rate(cls, currency, rate):
+        currency.sudo().rate_ids.unlink()
+        return cls.env['res.currency.rate'].create({
+            'name': '2019-01-01',
+            'rate': rate,
+            'currency_id': currency.id,
+            'company_id': cls.company_data['company'].id,
+        })
+
+    def _structure_move_vals(self, move_vals):
+        return {
```

Comment:
It is like you could just extend move_vals with company, partner_jo  (normally that is more the journal on invoices) put an invoice_date/currency if nothing is there, ...

Reply:
That would work if I did not need to create multiple lines (with possibly multiple taxes)